governance/default.go

```diff
@@ -667,18 +667,21 @@ func (g *Governance) initializeCache() error {
 		}
 	}
 
-	lastGovernanceBlock, lastGovernanceStateBlock := indices[len(indices)-1], atomic.LoadUint64(&g.lastGovernanceStateBlock)
-	if lastGovernanceBlock >= lastGovernanceStateBlock {
-		// head block number is used to get the appropriate g.currentSet and g.actualGovernanceBlock
-		headBlockNumber := uint64(0)
-		if headBlockHash := g.db.ReadHeadBlockHash(); !common.EmptyHash(headBlockHash) {
-			if num := g.db.ReadHeaderNumber(headBlockHash); num != nil {
-				headBlockNumber = *num
-			}
+	// head block number is used to get the appropriate g.currentSet and g.actualGovernanceBlock
+	headBlockNumber := uint64(0)
+	if headBlockHash := g.db.ReadHeadBlockHash(); !common.EmptyHash(headBlockHash) {
+		if num := g.db.ReadHeaderNumber(headBlockHash); num != nil {
+			headBlockNumber = *num
 		}
-		newBlockNumber, newGovernanceSet, _ := g.ReadGovernance(headBlockNumber)
-		g.currentSet.Import(newGovernanceSet)
-		g.actualGovernanceBlock.Store(newBlockNumber)
+	}
+	newBlockNumber, newGovernanceSet, _ := g.ReadGovernance(headBlockNumber)
+	g.actualGovernanceBlock.Store(newBlockNumber)
+	g.currentSet.Import(newGovernanceSet)
+
+	governanceBlock, governanceStateBlock := g.actualGovernanceBlock.Load().(uint64), atomic.LoadUint64(&g.lastGovernanceStateBlock)
+	if governanceBlock >= governanceStateBlock {
+		ret, _ := g.itemCache.Get(getGovernanceCacheKey(governanceBlock))
+		g.currentSet.Import(ret.(map[string]interface{}))
```

Comment:
Is this code safe when `ret` is `nil` ?

Reply:
Yes, It's not safe. I added error handling logic at https://github.com/klaytn/klaytn/pull/1168/commits/6a50c4107d7b5929eeb518f9728aa7c63aa12de4. Please take a look.